xla/service/gpu/conv_algorithm_picker.cc

```diff
@@ -675,9 +675,17 @@ absl::StatusOr<AutotuneResult> GpuConvAlgorithmPicker::AutotuneOneConvRunner(
 
   if (reference_result->has_value()) {
     XLA_SCOPED_LOGGING_TIMER_LEVEL("BufferComparator::CompareEqual", 2);
-    BufferComparator comparator(runtime_arguments.rz_buffers.output_shape(),
-                                runtime_arguments.hlo_module_config);
     for (int i = 0; i < result_buffers.size(); ++i) {
+      Shape output_shape;
+      if (runtime_arguments.rz_buffers.output_shape().IsTuple()) {
```

Comment:
Can you separate this if-else logic of calculating output shape to a separate function in anonymous namespace? Also comment on the behavior like e.g. here

https://github.com/openxla/xla/blob/53f1bd856cb9d25f895490ab793afe124304f457/xla/service/gpu/ir_emitter_triton.cc#L762-L798